### PR TITLE
Font color as function

### PR DESCRIPTION
**Pull request by Hintalo**
Added borderColor, hoverBackgroundColor and hoverBorderColor to the object argument that is passed to the fontColor callback function.
This allows more precise computation of the final text-color of the labels.

### CODE REVIEW

File: src/Chart.PieceLabel.js [hunk 1]
```diff
@@ -120,7 +120,10 @@
           value: dataset.data[i],
           percentage: percentage,
           text: text,
-          backgroundColor: dataset.backgroundColor[i]
+          backgroundColor: dataset.backgroundColor[i],
+          borderColor: dataset.borderColor ? dataset.borderColor[i] : null,
```
Comment (emn178) [inline src/Chart.PieceLabel.js]: or pass `dataset` and `index`?

**Comment by Hintalo** [comment]
Yes, good idea, I've changed the code in my pull request, please review again. Thx.

File: src/Chart.PieceLabel.js [hunk 2]
```diff
@@ -117,10 +117,10 @@
       if (typeof fontColor === 'function') {
         fontColor = fontColor({
           label: chartInstance.config.data.labels[i],
-          value: dataset.data[i],
           percentage: percentage,
           text: text,
-          backgroundColor: dataset.backgroundColor[i]
```
Comment (emn178) [inline src/Chart.PieceLabel.js]: please keep these lines. we can deprecate it but not breaking migration.

**Comment by Hintalo** [comment]
You are right, I've fixed that, so there are no breaking changes now.

**Pull request merged**